packages/panna-sdk/src/core/helpers/http.ts

```diff
@@ -0,0 +1,67 @@
+import * as axios from 'axios';
+import { newLruMemCache } from './cache';
+import { delay } from './delay';
+
+// Constants
+const DEFAULT_RETRIES = 0;
+const DEFAULT_RETRY_DELAY_MS = 100;
+
+// Request cache
+const requestCache = newLruMemCache('http_request');
+
+// Internal types
+type RequestParamsConfig = Omit<axios.AxiosRequestConfig, 'url'>;
+
+// Internal functions
+const isCacheableResponse = (response: axios.AxiosResponse) =>
+  response.status === HttpStatusCode.Ok;
+
+const requestWithRetries = async (
+  url: string,
+  requestParams: RequestParamsConfig,
+  numRetries: number = DEFAULT_RETRIES,
+  retryDelayMs: number = DEFAULT_RETRY_DELAY_MS
+) => {
+  let response: axios.AxiosResponse;
+
+  do {
+    const ax = new axios.Axios();
+    response = await ax.request({ url, ...requestParams });
+    if (response.status < HttpStatusCode.BadRequest) return response;
+
+    await delay(retryDelayMs);
```

Comment:
its better to have exponential backoff strategy that increases delay with each retry that gives server more time to recover
```
const delayMs = retryDelayMs * Math.pow(2, numRetries - 1);
await delay(retryDelayMs);
```